pkg/polaris/aws/aws.go

```diff
@@ -929,3 +929,148 @@ func (a API) TrustPolicies(ctx context.Context, id IdentityFunc, features []core
 
 	return trustPolicies, nil
 }
+
+func (a API) CreateCloudCluster(ctx context.Context, id uuid.UUID, input aws.CreateAwsClusterInput, useLatestCdmVersion bool) error {
+	a.log.Print(log.Trace)
+
+	// Ensure account exists and has Server and Apps feature
+	account, err := a.AccountByID(ctx, core.FeatureServerAndApps, id)
+	if err != nil {
+		return err
+	}
+
+	// If CloudAccountID is not specified, use the account ID from the account lookup
```

Comment:
I wonder if we should allow empty `CloudAccountID` in the input 🤔.
Regardless of if that is allowed, shouldn't we at least verify here that `input.CloudAccountID == account.ID` if it is provided?

Reply:
I think the better way to go here is to drop the `id` param altogether and we only use the input param instead with a `CloudAccountID` populated. I had done some refactoring whilst developing this and missed this change.